ymer/src/test/java/com/avanza/ymer/RepeatingTaskTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2015 Avanza Bank AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.avanza.ymer;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assume.assumeTrue;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.junit.Test;
+
+public class RepeatingTaskTest {
+
+	@Test
+	public void shouldTriggerExpectedNumberOfTimes() throws Exception {
+		LongAdder counter = new LongAdder();
+
+		try(RepeatingTask ignore = new RepeatingTask(Duration.ofMillis(5), Duration.ofMillis(10), counter::increment)) {
+			Thread.sleep(100);
+		}
+
+		assertThat(counter.sum(), is(10L));
+	}
+
+	@Test
+	public void shouldNotTriggerAfterClosed() throws Exception {
+		LongAdder counter = new LongAdder();
+
+		try(RepeatingTask ignore = new RepeatingTask(Duration.ZERO, Duration.ofMillis(100), counter::increment)) {
```

Comment:
maybe use larger interval than 100ms to make test more stable? sleep(5) + sleep(10) with some thread context switching might sum up to +100ms if we are unlucky and have a build server under high load

Reply:
I've loosened the timings a bit in https://github.com/AvanzaBank/ymer/pull/74/commits/b87f6421851adf2d546ad9ec31d91bfc3ea4dd0a